OSX/xcode/ags/plugin_registration.cpp

```diff
@@ -0,0 +1,34 @@
+//
+//  plugin_registration.cpp
+//  Shivah
+//
+//  Created by Nick Sonneveld on 26/06/2016.
+//
+//
+
+#include "plugin_registration.hpp"
+
+#include "agsplugin.h"
+//#include "example.h"
+
+typedef void (*t_engine_pre_init_callback)(void);
+extern void engine_set_pre_init_callback(t_engine_pre_init_callback callback);
+
+static void pl_register_builtin_plugin() {
+
+//  pl_register_builtin_plugin((InbuiltPluginDetails){
+//    .filename = "example",
```

Comment:
Um, Nick, you really tricked me here (so I deleted my previous answer).

The example you referenced is about **initializing** char[] variable, which is ofcourse valid.
But in this case you suggest assigning new value to char[] variable. In C++ that won't work.